### ISSUE TITLE
Watermark processor broken in Rails3 and Paperclip 2.3.3

### ISSUE THREAD

**Issue by exviva** [ng/paperclip-watermarking-app#1]
Hi Jonathan,

nice job with the watermark processor :).

I saw that your project is a bit unmaintained now, and I modified it a bit to work with Rails3/Paperclip 2.3.3.

Here's an article on what I did: http://exviva.posterous.com/watermarking-images-with-rails-3-and-papercli.

And here's the modified watermark.rb file:
http://gist.github.com/raw/465088/3de6e4f50ff659b91ee094764af842968d4e0de3/watermark.rb.

Let me know if there's any other way I could help.

**Comment by ng** [comment]
Just saw this, somehow it got lost in my spam folder! I'll check it out this evening.

**Comment by kkerley** [comment]
Thanks for your help updating this to Rails 3.  Sadly, it didn't work for me.  I posted the other issue on this plugin about how it works perfectly in development but as soon as it goes into production, all hell breaks loose.

It used to work where it would watermark one photo and then crap out on each subsequent attempt...now it just flat out doesn't work--using both your modification, exviva, as well as the one suggested in my issue about forcing the geometry in the plugin itself.

Are you running Passenger on your production server?  It's the only thing I can think of that would cause a problem.  I've frozen Rails (2.3.8) and all gems so the environment is identical otherwise.

**Comment by exviva** [comment]
@kkerley indeed the watermarking is fine in production.

We're using Rails 3, Passenger 2.2.15 (switched to Passenger 3 yesterday), nginx, Ubuntu Hardy, ImageMagick 6.3.7.

Have you tried running the ImageMagick commands directly in the shell? Maybe it's not the gem's fault after all?

**Comment by kkerley** [comment]
@exviva 

I haven't, no.  And I'm not sure what version of ImageMagick is on the server, actually.  It came stock from the hosting company (but it's a VPS and I have full root/sudo access to it) so maybe an upgrade there is in order?  I'll have to try it.  My Rails app isn't overly complex so I really don't think it's a problem with a lot of weird gem compatibilities or something.  It's gotta be something with the server environment...

**Comment by exviva** [comment]
@kkerley indeed, we have bumped into this issue as well, I just haven't noticed it!

We upgraded to Paperclip 2.3.5 (actually our own fork including jm81's fix and one or two of ours), and modified the watermark processor as described in issue 2.
